plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XWorkspaceBuilder.java

```diff
@@ -274,21 +299,27 @@ private BuildTask toBuildTask(WorkspaceChanges workspaceChanges) {
 			ProjectBuildOrderIterator pboIterator = projectBuildOrderInfo.getIterator(changedPDs);
 
 			while (pboIterator.hasNext()) {
+
 				ProjectDescription descr = pboIterator.next();
 				ProjectBuilder projectBuilder = workspaceManager.getProjectBuilder(descr.getName());
 				Set<URI> projectDirty = project2dirty.getOrDefault(descr, Collections.emptySet());
 				Set<URI> projectDeleted = project2deleted.getOrDefault(descr, Collections.emptySet());
 
-				// the projectResult might contain partial information in case the build was cancelled
 				XBuildResult projectResult;
+				AffectedResourcesRecordingFactory recordingFactory = new AffectedResourcesRecordingFactory(
+						buildRequestFactory);
 				try {
-					projectResult = projectBuilder.doIncrementalBuild(buildRequestFactory, projectDirty, projectDeleted,
-							allBuildDeltas, cancelIndicator);
-				} catch (BuildCanceledException e) {
-					// TODO GH-1793 remove this temporary solution
-					dirtyFiles.addAll(e.incompletelyBuiltFiles);
-					e.rethrowOriginalCancellation();
-					throw new IllegalStateException("should never get here");
+					projectResult = projectBuilder.doIncrementalBuild(
+							recordingFactory,
+							projectDirty,
+							projectDeleted,
+							allBuildDeltas,
```

Comment:
while at it, please consider renaming `allBuildDeltas` to `allBuiltDeltas` (I guess the name means it contains deltas that "were built")

Reply:
according to the JavaDoc these are deltas that have not been taken into account yet by downstream projects. I'll try to find a better name, since `all` is certainly a misleading prefix, too.

Reply:
`toBeConsideredDeltas`